Bottom: 510b3d95881375337a9bc3b0aaee65412494a4ce
Top:    b557c9e08c230c5c0bad3365324d25baea334e53
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 18:11:17 +0000

AddElementToBuffer(): Let CopyElementLowLevel() create the element to copy into.


---

diff --git a/src/buffer.c b/src/buffer.c
index 44b181c..5835788 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -211,11 +211,7 @@ AddPolygonToBuffer (LayerType *Layer, PolygonType *Polygon)
 static void *
 AddElementToBuffer (ElementType *Element)
 {
-  ElementType *element;
-
-  element = GetElementMemory (Dest);
-  CopyElementLowLevel (Dest, element, Element, false, 0, 0, FOUNDFLAG | ExtraFlag);
-  return (element);
+  return CopyElementLowLevel (Dest, NULL, Element, false, 0, 0, FOUNDFLAG | ExtraFlag);
 }
 
 /* ---------------------------------------------------------------------------
